src/Algebra/Morphism/Bundles.agda

```diff
@@ -0,0 +1,178 @@
+    isMonoidHomomorphism : IsMonoidHomomorphism A.rawMonoid B.rawMonoid ⟦_⟧
+
+  open IsMonoidHomomorphism isMonoidHomomorphism public
+
+  magmaHomomorphism : MagmaHomomorphism A.magma B.magma
+  magmaHomomorphism = record { isMagmaHomomorphism = isMagmaHomomorphism }
+
+  open MagmaHomomorphism magmaHomomorphism public
+
+------------------------------------------------------------------------
+-- Morphisms between Groups
+------------------------------------------------------------------------
+
+record GroupHomomorphism (A : Group a ℓa) (B : Group b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = Group A
+    module B = Group B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isGroupHomomorphism : IsGroupHomomorphism A.rawGroup B.rawGroup ⟦_⟧
+
+  open IsGroupHomomorphism isGroupHomomorphism public
+
+  monoidHomomorphism : MonoidHomomorphism A.monoid B.monoid
+  monoidHomomorphism = record { isMonoidHomomorphism = isMonoidHomomorphism }
+
+  open MonoidHomomorphism monoidHomomorphism public
+
+------------------------------------------------------------------------
+-- Morphisms between NearSemirings
+------------------------------------------------------------------------
+
+record NearSemiringHomomorphism (A : NearSemiring a ℓa) (B : NearSemiring b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = NearSemiring A
+    module B = NearSemiring B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isNearSemiringHomomorphism : IsNearSemiringHomomorphism A.rawNearSemiring B.rawNearSemiring ⟦_⟧
+
+  open IsNearSemiringHomomorphism isNearSemiringHomomorphism public
+
+  +-monoidHomomorphism : MonoidHomomorphism A.+-monoid B.+-monoid
+  +-monoidHomomorphism = record { isMonoidHomomorphism = +-isMonoidHomomorphism }
+
+  open MonoidHomomorphism +-monoidHomomorphism public
+
+  *-magmaHomomorphism : MagmaHomomorphism A.*-magma B.*-magma
+  *-magmaHomomorphism = record { isMagmaHomomorphism = *-isMagmaHomomorphism }
+
+------------------------------------------------------------------------
+-- Morphisms between Semirings
+------------------------------------------------------------------------
+
+record SemiringHomomorphism (A : Semiring a ℓa) (B : Semiring b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = Semiring A
+    module B = Semiring B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isSemiringHomomorphism : IsSemiringHomomorphism A.rawSemiring B.rawSemiring ⟦_⟧
+
+  open IsSemiringHomomorphism isSemiringHomomorphism public
+
+  *-monoidHomomorphism : MonoidHomomorphism A.*-monoid B.*-monoid
+  *-monoidHomomorphism = record { isMonoidHomomorphism = *-isMonoidHomomorphism }
+
+  open MonoidHomomorphism *-monoidHomomorphism public
+
+------------------------------------------------------------------------
+-- Morphisms between RingWithoutOnes
+------------------------------------------------------------------------
+{-
+
+Problem: bundle RingWithoutOne doesn't re-export its underlying RawRingWithoutOne!
```

Comment:
Seems like a bug that should be fixed within this PR as a way forward?

Reply:
Now added to `Algebra.Bundles.RingWithoutOne`.

Reply:
But still problems: see latest commits...
```agda
Cannot resolve overloaded projection Carrier because
it is not applied to a visible argument
  candidates in scope:
- Carrier : Set a
- Carrier : Set a
when checking that the expression
Algebra.Morphism.Bundles.A.Carrier has type _226
```
if declaring
```agda
field
    ⟦_⟧ : A.Carrier → B.Carrier
```
???

Reply:
Ditto. `rawKleeneAlgebra` added to `Algebra.Bundles.KleeneAlgebra`.